src/components/Transactions/TransactionGeneric.tsx

```diff
@@ -0,0 +1,174 @@
+  const fromAddress = txDetails.ensFrom || txDetails.from
+
+  return (
+    <>
+      <Flex
+        alignItems='center'
+        flex={1}
+        justifyContent='space-between'
+        textAlign='left'
+        as='button'
+        w='full'
+        py={4}
+        onClick={toggleOpen}
+      >
+        <Flex alignItems='center' width='full'>
+          <IconCircle mr={3}>
+            <FaStickyNote />
+          </IconCircle>
+
+          <Flex justifyContent='flex-start' flex={1} alignItems='center'>
+            <Box flex={1}>
+              <Text
+                fontWeight='bold'
+                overflow='hidden'
+                flex={1}
+                textOverflow='ellipsis'
+                maxWidth='60%'
+                lineHeight='1'
+                whiteSpace='nowrap'
+                mb={2}
+                translation={['transactionRow.unknown', { symbol: '' }]}
+              />
+              <RawText color='gray.500' fontSize='sm' lineHeight='1'>
+                {dayjs(txDetails.tx.blockTime * 1000).fromNow()}
+              </RawText>
+            </Box>
+
+            <Flex flexDir='column' ml='auto' textAlign='right'>
+              {txDetails.value && txDetails.precision && (
+                <Amount.Crypto
+                  color='inherit'
+                  value={fromBaseUnit(txDetails.value, txDetails.precision)}
+                  symbol={txDetails.symbol}
+                  maximumFractionDigits={6}
+                  prefix=''
+                />
+              )}
+            </Flex>
+          </Flex>
+        </Flex>
+      </Flex>
+      <Collapse in={isOpen} unmountOnExit>
+        <SimpleGrid gridTemplateColumns='repeat(auto-fit, minmax(180px, 1fr))' spacing='4' py={6}>
+          {txDetails.tx.blockTime && (
+            <Row variant='vertical' hidden={!txDetails.tx.blockTime}>
+              <Row.Label>
+                <Text translation='transactionRow.date' />
+              </Row.Label>
+              <Row.Value>{dayjs(Number(txDetails.tx.blockTime) * 1000).format('LLL')}</Row.Value>
+            </Row>
+          )}
+          <Row variant='vertical' hidden={!(txDetails.explorerTxLink && txDetails.tx.txid)}>
+            <Row.Label>
+              <Text translation='transactionRow.txid' />
+            </Row.Label>
+            <Row.Value>
+              <Link
+                isExternal
+                color='blue.500'
+                href={`${txDetails.explorerTxLink}${txDetails.tx.txid}`}
+              >
+                <MiddleEllipsis address={txDetails.tx.txid} />
+              </Link>
+            </Row.Value>
+          </Row>
+
+          <Row variant='vertical' hidden={!txDetails.tradeTx}>
+            <Row.Label>
+              <Text translation={'transactionRow.amount'} />
+            </Row.Label>
+            <Row.Value>
+              <Amount.Crypto
+                value={fromBaseUnit(
+                  txDetails.sellTx?.value ?? '0',
```

Comment:
Would it make sense to accept `BigNumber.Value | null | undefined` as a value in `fromBaseUnit`? 
https://github.com/shapeshift/web/blob/433be3392ff233bdaf1141f73d44fe67f68aef09/src/lib/math.ts#L6

Since we're using `bnOrZero`, IMO we should just use the same input types that it accepts:
https://github.com/shapeshift/web/blob/433be3392ff233bdaf1141f73d44fe67f68aef09/src/lib/bignumber/bignumber.ts#L8

This would allow us to avoid using these kind of fallbacks